### DISCUSSION TITLE
`impl Deserialize` uses `deserializer.deserialize_bytes`, unsupported by `serde_ya?ml`

### DISCUSSION THREAD

**Discussion by LeoniePhiline** [BurntSushi/jiff#148] [Q&A] [answered]
Hi!

Deserializing e.g. a `jiff::Span` from an ISO duration string in a YAML file is not currently possible.

Neither `serde_yaml` nor its maintained fork `serde_yml` support deserializing from bytes:

https://docs.rs/crate/serde_yml/latest/source/src/de.rs#1908-1920

Error: https://docs.rs/crate/serde_yml/latest/source/src/modules/error.rs#125

What could be the best way forward?

Could `impl<'de> serde::Deserialize<'de> for Span` possibly switch to `deserializer.deserialize_str`, or is that out of the question?

The same applies to `impl Deserialize for ...`: `Zoned`, `Timestamp`, `SignedDuration`, `civil::Date`, `civil::Time` and `civil::DateTime`.

**Comment by BurntSushi** [comment, marked as answer]
I _think_ this question is actually an answer to the questions I asked in #138. Nice find. In #138, there is a suggestion to use `deserialize_str`, and my question was basically, "well, is the current implementation causing issues?" And it seems like it is.

I would accept a PR switching everything over to `deserialize_str`. (Or I'll do it when I get a chance.)

Thank you for bringing this up!

**Reply by LeoniePhiline** [reply]
You will find my PR at https://github.com/BurntSushi/jiff/pull/149.

**Reply by LeoniePhiline** [reply]
Tests fail unexpectedly. I must have missed a difference between my local machine and CI. I will let you know when tests pass.

**Reply by LeoniePhiline** [reply]
Tests at #149 succeed now.

Cross-compilation CI jobs had no access to the timezone database; I have thus switched the added tests to using `UTC` instead.

Please note that the failure of [ci / wasm32-unknown-uknown (pull_request)](https://github.com/BurntSushi/jiff/actions/runs/11615853944/job/32347436795?pr=149) is unrelated. 

These seem to [fail in all cases](https://github.com/BurntSushi/jiff/actions) since October 17th.
